src/main/java/io/stargate/sgv2/jsonapi/service/updater/DocumentUpdater.java

```diff
@@ -105,9 +132,65 @@ private boolean replace(ObjectNode docToUpdate, boolean docInserted) {
     return true;
   }
 
+  /**
+   * vectorize replacementDocument as needed, only when document is found will be used by
+   * findOneAndReplace
+   *
+   * @param dataVectorizer
+   */
+  public Uni<Boolean> vectorizeTheReplacementDocument(DataVectorizer dataVectorizer) {
+    return dataVectorizer.vectorize(List.of(replaceDocument));
+  }
+
+  /**
+   * Check if there is any $vectorize diff If there are docs found to update or doc to replace, then
+   * this is a necessary condition to proceed vectorization
+   *
+   * @param foundDocs
+   */
+  public boolean hasVectorizeDiff(List<ReadDocument> foundDocs) {
```

Comment:
per above, some of this logic is also outside in the ReadAndUpdateOpeation , we should encapsulate all the logic in this function